### PR TITLE
GCed ref causes flaky test

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java [hunk 1]
```diff
@@ -338,6 +338,7 @@ public void findFirstAsync() {
                 .between(AllTypes.FIELD_LONG, 4, 9)
                 .findFirstAsync();
         assertFalse(allTypes.isLoaded());
+        looperThread.keepStrongReference.add(allTypes);
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java]: If this indeed fixes its we should have a ton of bugs in all other unit tests using the `@RunTestInLooperThread`?
`allTypes` do go out of scope when exiting the method though 😢

**Comment by stk1m1** [comment]
👍 as soon as Mr. Jenkins says ok.

**Comment by zaki50** [comment]
:+1:

File: realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java (hunk 1 above)
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java]: I think I fixed most of them last time. So there should be quite a few left.

**Pull request merged**